File: Backend/services/stakeholders/dto/user_dto.go
package dto

import (
	"time"

	"github.com/google/uuid"
)

type UserUpdateRequest struct {
	Description  string    `json:"description" gorm:"primaryKey"`
	FirstName    string    `json:"FirstName" gorm:"primaryKey"`
	LastName     string    `json:"LastName" gorm:"primaryKey"`
	Moto         string    `json:"Moto" gorm:"primaryKey"`
	ProfilePhoto string    `json:"ProfilePhoto" gorm:"primaryKey"`

}

type UserResponse struct {
	Id 			 uuid.UUID `json:"id" gorm:"primaryKey"`
	Username     string    `json:"username" gorm:"primaryKey"`
	Email        string    `json:"email" gorm:"primaryKey"`
	PasswordHash string    `json:"passwordHash" gorm:"primaryKey"`
	Role         string    `json:"role" gorm:"primaryKey"`
	CreatedAt    time.Time `json:"createdAt" gorm:"primaryKey"`
	UpdatedAt    time.Time `json:"updatedAt" gorm:"primaryKey"`
	Description  string    `json:"description" gorm:"primaryKey"`
	FirstName    string    `json:"FirstName" gorm:"primaryKey"`
	LastName     string    `json:"LastName" gorm:"primaryKey"`
	Moto         string    `json:"Moto" gorm:"primaryKey"`
	ProfilePhoto string    `json:"ProfilePhoto" gorm:"primaryKey"`
}

File: Backend/services/auth/AuthService/DTOs/UserDto.cs
using AuthService.Models;

namespace AuthService.DTOs {
    public class UserDto {
        public int Id { get; set; }
        public string Username { get; set; }
        public string Email { get; set; }
        public string Role { get; set; }
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public string? Description { get; set; }
        public string? Moto { get; set; }
        public string? ProfilePhoto { get; set; }
        public UserStatus Status { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
        public DateTimeOffset? BlockedAt { get; set; }
    }

}
